core/include/engine/Hamiltonian_Heisenberg.hpp

```diff
@@ -105,6 +107,10 @@ namespace Engine
         scalarfield ddi_magnitudes;
         vectorfield ddi_normals;
 
+	    // ------------ Triplet Interactions ------------
+		tripletfield triplets;
```

Comment:
It seems you used tabs for indentation sometimes - please use spaces everywhere (FYI: VS Code can show you which you are using and also replace them for you)

Reply:
I added the CUDA backend (didn't check for correctness though) and fixed the indentation.